completed section js: me quede por handling errors (section19) y pase a async away para luego volver
